account_financial_risk/models/res_company.py

```diff
@@ -13,3 +13,11 @@ class ResCompany(models.Model):
              "The change of this field recompute all partners risk,"
              "be patient.",
     )
+
+    # Remove when odoo compare values before to write
+    def write(self, vals):
+        new_margin = vals.get('invoice_unpaid_margin')
+        if (new_margin is not None and
+                new_margin == self[:1].invoice_unpaid_margin):
```

Comment:
You should check for each of the written companies. `self` is usually one, but no one knows...:
```suggestion
                all([new_margin == x.invoice_unpaid_margin for x in self]):
```

Reply:
Done